pkg/configdb/configdb.go

```diff
@@ -0,0 +1,77 @@
+package configdb
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path"
+	"sync"
+)
+
+var mu sync.Mutex
+
+type GlobalOptions struct {
+	IP                 string
+	Gateway            string
+	LeaseDuration      int
+	AuthToken          string
+	ListenInterface    string
+	HTTPListenAddress  string
+	TLSEnabled         bool
+	TLSPrivateKeyFile  string
+	TLSCertificateFile string
+	Options            interface{}
+}
+
+type Clients struct {
+	Clients []Client
+}
+
+type Client struct {
+	Hostname string
+	MAC      string
+	IP       string
+	Options  interface{}
+}
+
+var DB *Clients
+var Config *GlobalOptions
+
+func Init(config_path string) error {
+	content, err := os.ReadFile(path.Join(config_path, "rest-dhcpd-clients.json"))
+	if err != nil {
+		log.Printf("%s", err)
+		return err
+	}
+	err = json.Unmarshal(content, &DB)
+	if err != nil {
+		log.Printf("%s", err)
+		return err
+	}
+
+	cfg, err := os.ReadFile(path.Join(config_path, "rest-dhcpd-config.json"))
+	if err != nil {
+		log.Printf("%s", err)
+		return err
+	}
+	err = json.Unmarshal(cfg, &Config)
+	if err != nil {
+		log.Printf("%s", err)
+		return err
+	}
+	log.Printf("DB init done.")
+	return nil
+}
+
+func Save() {
+	content, err := json.MarshalIndent(DB, "", "  ")
+	if err != nil {
+		log.Printf("%s", err)
+	}
+	mu.Lock()
+	err = os.WriteFile("rest-dhcp-clients.json", content, 0644)
+	mu.Unlock()
+	if err != nil {
+		log.Printf("%s", err)
```

Comment:
Please return `error`from this function and handle it in the caller.

Reply:
fixed